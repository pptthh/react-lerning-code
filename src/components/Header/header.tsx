import * as React from 'react';

interface HeaderProps {
    children: any;
    tooltip?: string;
}

const Header = ({children, tooltip = ''}: HeaderProps): JSX.Element => 
    <h5 className="header" title={tooltip}>{children}</h5>;

export default Header;
