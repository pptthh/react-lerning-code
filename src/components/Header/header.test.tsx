import * as React from 'react';
import Header from './header';
import toJson from 'enzyme-to-json';
import * as ReactSixteenAdapter from 'enzyme-adapter-react-16';
import {configure, render, shallow} from 'enzyme';

configure({adapter: new ReactSixteenAdapter()});

it('renders correctly with Header', () => {
    const tree = render(<Header>Panel Header</Header>);

    expect(toJson(tree)).toMatchSnapshot();
});

it('Shallow render example', () => {
    const tree = shallow(<h1><Header tooltip={'tooltip'}>Panel Header in H1</Header></h1>);

    expect(toJson(tree)).toMatchSnapshot();
});
