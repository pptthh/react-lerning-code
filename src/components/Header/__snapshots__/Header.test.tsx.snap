// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shallow render example 1`] = `
<h1>
  <Header
    tooltip="tooltip"
  >
    Panel Header in H1
  </Header>
</h1>
`;

exports[`renders correctly with Header 1`] = `
<h5
  class="header"
  title=""
>
  Panel Header
</h5>
`;
